source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/MediaRenderer.java

```diff
@@ -0,0 +1,349 @@
+                }
+
+                m_videoView.start();
+                m_videoView.setVisibility(View.VISIBLE);
+                m_playButon.setVisibility(View.GONE);
+            }
+
+            m_cardActionHandler.onMediaPlay(m_cardMediaElement, m_renderedAdaptiveCard);
+        }
+
+        private ImageView m_poster;
+        private ImageView m_playButon;
+        private VideoView m_videoView;
+        private boolean m_allowedInlinePlayback = true;
+
+        private BaseCardElement m_cardMediaElement;
+        private RenderedAdaptiveCard m_renderedAdaptiveCard;
+        private ICardActionHandler m_cardActionHandler;
+    }
+
+    private class MediaOnCompletionListener implements MediaPlayer.OnCompletionListener
+    {
+
+        MediaOnCompletionListener(BaseCardElement mediaElement, RenderedAdaptiveCard renderedAdaptiveCard, ICardActionHandler cardActionHandler)
+        {
+            m_cardMediaElement = mediaElement;
+            m_renderedAdaptiveCard = renderedAdaptiveCard;
+            m_cardActionHandler = cardActionHandler;
+        }
+
+        @Override
+        public void onCompletion(MediaPlayer mp)
+        {
+            m_cardActionHandler.onMediaStop(m_cardMediaElement, m_renderedAdaptiveCard);
+        }
+
+        private BaseCardElement m_cardMediaElement;
+        private RenderedAdaptiveCard m_renderedAdaptiveCard;
+        private ICardActionHandler m_cardActionHandler;
+    }
+
+    private class MediaOnPreparedListener implements MediaPlayer.OnPreparedListener
+    {
+        public MediaOnPreparedListener(MediaController mediaController)
+        {
+            m_mediaController = mediaController;
+        }
+
+        @Override
+        public void onPrepared(MediaPlayer mp)
+        {
+            m_mediaController.show(0);
+            m_mediaController.requestLayout();
+        }
+
+        private MediaController m_mediaController;
+    }
+
+    class OnlineFileCheckerTask extends AsyncTask<String, Void, Boolean>
+    {
+
+        protected Boolean doInBackground(String... url)
+        {
+            try
+            {
+                HttpURLConnection connection = (HttpURLConnection)new URL(url[0]).openConnection();
+                connection.setRequestMethod("HEAD");
+                return (connection.getResponseCode() == HttpURLConnection.HTTP_OK);
+            }
+            catch (Exception e)
+            {
+                return false;
+            }
+        }
+
+    }
+
+    private ImageView renderPoster(
+            RenderedAdaptiveCard renderedCard,
+            Context context,
+            FragmentManager fragmentManager,
+            ViewGroup viewGroup,
+            Media media,
+            ICardActionHandler cardActionHandler,
+            HostConfig hostConfig,
+            ContainerStyle containerStyle)
+    {
+        ImageView posterView = null;
+        if(!(media.GetPoster().isEmpty()) || !(hostConfig.getMedia().getDefaultPoster().isEmpty()))
```

Comment:
rather than having to duplicate these checks below for assignment, should we instead just check to see if `poster`'s URL is set?